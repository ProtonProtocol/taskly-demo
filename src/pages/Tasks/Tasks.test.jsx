import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Tasks from './Tasks';

const createTestProps = props => ({
  accountData: {
    name: 'test',
    avatar: '',
  },
  logout: jest.fn(),
  ...props,
});

describe('Tasks component', () => {
  let props;

  beforeEach(() => {
    props = createTestProps({});
  });

  it('renders without crashing', () => {
    shallow(<Tasks {...props} />);
  });

  it('matches the rendered snapshot', () => {
    const component = shallow(<Tasks {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
