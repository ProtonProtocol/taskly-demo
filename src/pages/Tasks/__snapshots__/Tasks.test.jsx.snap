// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks component matches the rendered snapshot 1`] = `
<LoggedInLayout
  avatar=""
  logout={[MockFunction]}
  mainImageUrl="./images/smiling-man-bust.png"
  openConfirmModal={null}
>
  <h1>
    Thanks, now let's get started!
  </h1>
  <div
    className="description"
  >
    What can we help you with? Add the tasks below that you would like help with.
  </div>
  <input
    className="task-item"
    placeholder="New Task"
  />
  <div
    className="description add-task"
  >
    <span>
      Add another task
    </span>
    <PlusIcon />
  </div>
  <RedButton
    className="finish-button"
    handleOnClick={[Function]}
    isLoading={false}
    text="Finish"
  />
</LoggedInLayout>
`;
